src/pages/RecipeDetail.js

```diff
@@ -1,14 +1,36 @@
+import {useState} from 'react';
+import {AiTwotoneDelete} from 'react-icons/ai';
 import {BiCategoryAlt} from 'react-icons/bi';
 import {BsClockHistory} from 'react-icons/bs';
 import {FiAlignJustify} from 'react-icons/fi';
-import {useParams} from 'react-router-dom';
+import {useParams, useNavigate} from 'react-router-dom';
 import styled from 'styled-components';
 
-export default function RecipeDetail({recipes}) {
+import Dialogue from '../components/Dialogue/Dialogue';
+
+export default function RecipeDetail({recipes, onDeleteRecipe}) {
   const params = useParams();
+  const navigate = useNavigate();
   const recipe = recipes.find(recipe => recipe.id === params.id);
   const ingredientItems = recipe.ingredients.map((ingredient, index) => <li key={index}>{ingredient}</li>);
 
+  const [isDialogueShown, setIsDialogueShown] = useState(false);
+
+  const showDialogue = () => {
+    setIsDialogueShown(true);
+    console.log(isDialogueShown);
+  };
+
+  const hideDialogue = () => {
+    setIsDialogueShown(false);
+    console.log(isDialogueShown);
+  };
```

Comment:
You could actually combine these two in one function though:
```suggestion
  const toggleDialogue = () => {
    setIsDialogueShown(!isDialogueShown);
  };
```